compiler/noirc_printable_type/src/lib.rs

```diff
@@ -311,29 +311,12 @@ pub fn decode_printable_value<F: AcirField>(
                 .expect("not enough data to decode variable array length")
                 .to_u128() as usize;
 
-            let flattened_capacity = field_iterator
-                .next()
-                .expect("expected slice to have at least 1 field for capacity")
-                .to_u128() as usize;
-
-            // Capacity is in terms of flattened members. To figure out how many items are
-            // in the slice, we have to divide that with how wide the type is.
-            let capacity = match printable_type_width(typ) {
-                0 => length,
-                n => flattened_capacity / n,
-            };
-
             let mut array_elements = Vec::with_capacity(length);
 
             for _ in 0..length {
                 array_elements.push(decode_printable_value(field_iterator, typ));
             }
 
-            // Consume padding.
```

Comment:
The interpreter needs to be updated as well, in `intrinsics.rs`, not to append the capacity.